### PR TITLE
Fix failure when creating a schema with non-lowercase username on Iceberg Glue catalog

### LINKED ISSUE #16116: Iceberg Glue catalog throws "Explicit schema owner is not supported" when username contains uppercase characters
I'm facing the error: `Explicit schema owner is not supported` 
when i try to execute the command:  `CREATE SCHEMA  iceberg.trino_test5;`

I saw a detail:  when the logged user name contains ANY upper case letter the error is thrown. If all the user name letters are lower case, the error isn't triggered.

Looking through the source code I realized that the error is triggered in this snippet:

`
...
@Override
    public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
    {
        checkArgument(owner.getType() == PrincipalType.USER, "Owner type must be USER");
        checkArgument(owner.getName().equals(session.getUser()), "Explicit schema owner is not supported");
`
From the class io.trino.plugin.iceberg.catalog.glue.TrinoGlueCatalog.
This classes uses the constructor: 
`

@JsonCreator
    public TrinoPrincipal(@JsonProperty("type") PrincipalType type, @JsonProperty("name") String name)
    {
        this.type = requireNonNull(type, "type is null");
        this.name = name.toLowerCase(ENGLISH);
    }
`

from  io.trino.spi.security.TrinoPrincipal class. 

I suspect that ` name.toLowerCase(ENGLISH);`  snnipet can be reason of the problem.

[TrinoGlueCreateSchemaException.txt](https://github.com/trinodb/trino/files/10736415/TrinoGlueCreateSchemaException.txt)

I'm using the Iceberg connector and the glue catalog.

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Don't lowercase `name` in `TrinoPrincipal` when `PrincipalType` is `USER`. 
Fixes #16116

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Iceberg
* Fix failure when creating a schema with a username containing uppercase characters on Iceberg Glue catalog. ({issue}`16116`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
CI hit #16137

**Review by krvikash — APPROVED** [review]
LGTM. 
`Fixes https://github.com/trinodb/trino/issues/16116` in the description.

**Comment by dain** [comment]
I'm not sure of the impact of this change, and this impacts security commands, so I'm not comfortable approving right now.  

The `USER` string is an `identifier` in SQL, and I believe these have canonicalization rules covered by the SQL spec. I believe Trino is adding support for this in some cases, and this change should be in line with the other changes, so I added @martint as a reviewer.

**Comment by martint** [comment]
Per the SQL spec, identifiers should only be canonicalized (to upper-case) when they are not "delimited" (i.e., in double quotes). Obviously, making such a change wholesale would break compatibility. Also, many systems have different rules. Therefore, the plan so far is to let plugins decide how to canonicalize identifiers (schemas, tables, columns, users, roles, etc) before the values are used in other connector/plugin API calls.

The change proposed in this PR, as it stands, may break compatibility for plugins that are case-insensitive.

**Comment by electrum** [comment]
Should we instead fix this for Glue by lowercasing the name in `TrinoGlueCatalog`?

**Comment by findinpath** [comment]
@ebyhr  can you pls rebase to address the conflicts?

**Comment by krvikash** [comment]
Hi @ebyhr, Is anything remaining in this PR?

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java [hunk 1]
```diff
@@ -253,7 +253,7 @@ public Optional<TrinoPrincipal> getNamespacePrincipal(ConnectorSession session,
     public void createNamespace(ConnectorSession session, String namespace, Map<String, Object> properties, TrinoPrincipal owner)
     {
         checkArgument(owner.getType() == PrincipalType.USER, "Owner type must be USER");
-        checkArgument(owner.getName().equals(session.getUser()), "Explicit schema owner is not supported");
+        checkArgument(owner.getName().equalsIgnoreCase(session.getUser()), "Explicit schema owner is not supported");
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java:256]: why would owner name get lowercased?

why not compare equality with `session.getUser().toLower`?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogConnectorSmokeTest.java [hunk 2]
```diff
@@ -197,6 +199,20 @@ public void testCommentViewColumn()
         }
     }
 
+    @Test
+    public void testCreateSchemaWithNonLowercaseOwnerName()
+    {
+        Session newSession = Session.builder(getSession())
+                .setIdentity(Identity.ofUser("User"))
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/catalog/glue/TestIcebergGlueCatalogConnectorSmokeTest.java:206]: Should we have a reusable (connector) (smoke) test?

**Comment by ebyhr** [comment]
/test-with-secrets sha=25760946b0a17393dcf1f1f6cdc0a6bda3fce2f4

**Review by electrum — APPROVED** [review]

**Pull request merged**